app/src/main/java/com/example/triptracker/view/profile/UserProfileEditScreen.kt

```diff
@@ -0,0 +1,526 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.compose.rememberNavController
+import coil.compose.AsyncImage
+import com.example.triptracker.model.profile.UserProfile
+import com.example.triptracker.view.Navigation
+import com.example.triptracker.view.NavigationBar
+import com.example.triptracker.view.theme.Montserrat
+import com.example.triptracker.view.theme.md_theme_grey
+import com.example.triptracker.view.theme.md_theme_light_dark
+import com.example.triptracker.view.theme.md_theme_light_error
+import com.example.triptracker.view.theme.md_theme_orange
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.Date
+
+/**
+ * This function retrieves the user profile from the database.
+ *
+ * @return user profile.
+ */
+private fun retrieveProfile(): UserProfile {
+  // TODO: implement the retrieval of the ambient user profile
+  return UserProfile(
+      "jean.rousseau@epfl.ch",
+      "Jean-Jacques",
+      "Rousseau",
+      Date(),
+      "DarkSkyMan",
+      "profileImageUrl",
+      listOf(),
+      listOf())
+}
+
+/**
+ * This function saves the user profile to the database.
+ *
+ * @param profile : user profile to be saved.
+ */
+private fun updateProfile(
+    profile: UserProfile,
+    name: String,
+    surname: String,
+    mail: String,
+    birthdate: LocalDate,
+    username: String,
+    imageUrl: String?
+) {
+  // TODO: Implement saving the profile to the database
+}
+
+/**
+ * This composable function displays the user profile edit screen.
+ *
+ * @param navigation : Navigation object that manages the navigation in the application.
+ */
+@Composable
+fun UserProfileEditScreen(navigation: Navigation) {
+  val profile = retrieveProfile()
+
+  /* Mutable state variable that holds the name of the user profile */
+  var name by remember { mutableStateOf(profile.name) }
+  var isNameEmpty by remember { mutableStateOf(profile.name.isEmpty()) }
+
+  /* Mutable state variable that holds the surname of the user profile */
+  var surname by remember { mutableStateOf(profile.surname) }
+  var isSurnameEmpty by remember { mutableStateOf(profile.surname.isEmpty()) }
+
+  /* Mutable state variable that holds the mail of the user profile */
+  var mail by remember { mutableStateOf(profile.mail) }
+  var isMailEmpty by remember { mutableStateOf(profile.mail.isEmpty()) }
+
+  /* Mutable state variable that holds the birthdate of the user profile */
+  var birthdate by remember { mutableStateOf(LocalDate.now()) }
+  var isBirthdateEmpty by remember { mutableStateOf(false) }
+
+  /* Mutable state variable that holds the username of the user profile */
+  var username by remember {
+    mutableStateOf(profile.pseudo)
+  } // TODO: change when pseudo ---> username
+  var isUsernameEmpty by remember { mutableStateOf(profile.pseudo.isEmpty()) }
```

Comment:
The PR with this change has already been added to the master branch so we can already modify it

Reply:
All good !